### DISCUSSION TITLE
Midi output isn't accurate

### DISCUSSION THREAD

**Discussion by thejinkazama** [svsdval/video2midi#45] [Q&A]
Hello,

Thanks for this amazing work, I will donate once I get it working fully.

I'm trying to create a midi file out of this video [https://www.youtube.com/watch?v=LHYqMw9p4G4&t=84s](url).

The closest I can get is this: [https://file.io/X4olklze6dmF](url). As you can see/hear, it contains lots of random notes at the start and the keys are wrong (it plays an F to begin rather than an A) but the general tune is there.

I've also attached a few screenshots to show the settings and the keys being pressed when and where they should (I think). The first screenshot also shows lots of keys being pressed at the very start of the video (I don't understand why). 

Thank you so much for your help in getting this working!

![1](https://github.com/svsdval/video2midi/assets/150602709/5717dd97-80ab-43d2-b078-fb02f27d1b2f)
![2](https://github.com/svsdval/video2midi/assets/150602709/09debc5c-ee2b-4577-91c5-12952d522d2b)
![3](https://github.com/svsdval/video2midi/assets/150602709/b5ea5290-562a-49fb-b371-95aac285736b)

**Comment by C0rn3j** [comment]
Urm, did you happen to place those notes manually instead of scaling the default preset and repositioning it in its entirety?

**Comment by thejinkazama** [comment]
Hello,

Partially, yes. Everytime I tried to use the keyboard and mouse to
reposition it, there were some notes which needed manually adjusted. I also
had 3 extra notes which can be seen at the right above the keyboard.

Thanks for getting back to me,

Paul

**Reply by thejinkazama** [reply]
I think I may have found the problem but not the solution. 

The video above shows the keyboard starting on F, not C therefore the notes are being assigned to the wrong place. This is happening as it's impossible to set up the keyboard by right clicking on where the first C should be. 

Is there a way to delete the notes I don't need so the keyboard starts with F as in the video?

Thanks

**Comment by svsdval** [comment]
> The first screenshot also shows lots of keys being pressed at the very start of the video (I don't understand why).

Your sensitivity for dark colors is too high, because of this the app thinks that keys are pressed. If you need notes in dark cases, it is better to reduce the sensitivity and then add more colors to the palette by associating them with the desired channels.

Also instead of moving all the keys individually manually, use right mouse button to grab and move entire keyboard from specified (selected) key. The keys may extend beyond the boundaries of the screen - this is normal, in this case the app will ignore them.

checkout:
[Enclosure (Sniper Wolf Death song) - Metal Gear Solid ｜ Tutorial #1 [LHYqMw9p4G4].zip](https://github.com/svsdval/video2midi/files/13343917/Enclosure.Sniper.Wolf.Death.song.-.Metal.Gear.Solid.Tutorial.1.LHYqMw9p4G4.zip)

**Comment by thejinkazama** [comment]
Hello,

Thanks so much for all the help. I reduced the precolor sensitivity but it's still playing a huge array of notes at the very start and end of the song but never at any other point! This isn't an issue as I can simply skip this part with Synthesia.

In order to get the keyboard working based on the video size, I dragged all the keys which aren't visible in the octave containing C1 to the far right of the screen and moved F to the far left. 

This is now working perfectly - thanks again! Is this link still valid for donating?

[http://paypal.me/svsdval](url)

Paul

**Reply by svsdval** [reply]
> I reduced the precolor sensitivity but it's still playing a huge array of notes at the very start and end of the song but never at any other point! This isn't an issue as I can simply skip this part with Synthesia.

use buttons "set start frame" (hot key: s)  and  "set end frame (hot key: e)" to clip the video, all frames not included in the specified interval will be ignored.

> Is this link still valid for donating?

Alas, I don’t know, the link seems to be working, but it’s not clear whether PayPal will block it or whether it will accept donations. Until things get settled, I think it's better to donate to some other open source project.
